compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/AnalyserUtils.java

```diff
@@ -109,6 +110,13 @@ public static String getTypeDescription(TypeSymbol paramType) {
                     .filter(e -> !e.isEmpty() && !e.isBlank())
                     .reduce((a, b) -> String.join("|", a, b)).orElse("");
             return optionalSymbolAvailable ? concatenatedTypeDesc + Constants.OPTIONAL : concatenatedTypeDesc;
+        } else if (TypeDescKind.INTERSECTION.equals(paramKind)) {
+            List<TypeSymbol> availableTypes = ((IntersectionTypeSymbol) paramType).memberTypeDescriptors();
+            return availableTypes.stream()
+                    .filter(e -> TypeDescKind.TYPE_REFERENCE.equals(e.typeKind()))
+                    .map(AnalyserUtils::getTypeDescription)
+                    .filter(e -> !e.isEmpty() && !e.isBlank())
+                    .reduce((a, b) -> String.join("&", a, b)).orElse("");
```

Comment:
Why do we have to do string manipulations here? Can't we use the semantic node and check the types?